.github/workflows/my-cd-pipeline.yml

```diff
@@ -0,0 +1,30 @@
+name: Continuous Deployment
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'README.md'
+
+jobs:
+  build:
+    name: Deploy into production
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: login to dockerhub
+      run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
+    - name: build production docker image
+      run: docker build --target production --tag ${{ secrets.DOCKER_LOGIN }}/todo-app-production:${GITHUB_SHA::6} .
```

Comment:
Nice use of the commit tag to version them, but also note that later (when we switch to Azure) we'll rely on the `:latest` tag being set as well, so it might be worth tagging twice (you can do it in the same command, `docker build --tag tag-1 --tag tag2 ...`) & pushing both tags.

That's typical because it's useful to be able to grab specific versions, but often we just want the latest stable version, so that we pick up patches etc. for free